src/plugins/workspace/public/components/workspace_form/utils.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DEFAULT_CHECKED_FEATURES_IDS } from '../../../common/constants';
+
+import { WorkspaceFeature, WorkspaceFeatureGroup, WorkspaceFormErrors } from './types';
+
+export const isWorkspaceFeatureGroup = (
+  featureOrGroup: WorkspaceFeature | WorkspaceFeatureGroup
+): featureOrGroup is WorkspaceFeatureGroup => 'features' in featureOrGroup;
+
+export const isDefaultCheckedFeatureId = (id: string) => {
+  return DEFAULT_CHECKED_FEATURES_IDS.indexOf(id) > -1;
+};
+
+export const appendDefaultFeatureIds = (ids: string[]) => {
+  // concat default checked ids and unique the result
+  return Array.from(new Set(ids.concat(DEFAULT_CHECKED_FEATURES_IDS)));
+};
+
+export const isValidNameOrDescription = (input?: string) => {
+  if (!input) {
+    return true;
```

Comment:
return false; ? Treat empty inputs as invalid?

Reply:
Good point! I think this is for bypass description input. Empty is invalid for name, will fix later.